lifecycle/lifecycle/job/registry.py

```diff
@@ -124,7 +126,7 @@ def sync_registry_jobs(config: Config, plugin_engine: PluginEngine):
         cluster_jobs = _generate_job_map(list_cluster_jobs(config, plugin_engine))
         registry_jobs = _generate_job_map(list_job_registry(config))
         logger.debug(f'Found {len(cluster_jobs)} jobs in the cluster, {len(registry_jobs)} in the database')
-        job_status_count: Dict[str, int] = defaultdict(int)
+        job_status_count: dict[str, int] = defaultdict(int)
```

Comment:
Why this change? It doesn't change anything and you have to write an additional import for it to work, and since you are still importing `Dict` from `typing` (used at the bottom of this file) this change seems even more meaningless.

Reply:
Python 3.9 introduced this https://peps.python.org/pep-0585/. Starting from Python 3.9, using the generic types is now preferred rather than their `typing` counterpart. 
So I keep replacing it once I see it somewhere around.

Reply:
As you said, there is still `typing.Dict` import, so perhaps I should replace all occurrences in the whole file to get rid of it.